webapp/javascript/components/ExportData.tsx

```diff
@@ -82,6 +112,52 @@ function ExportData(props: ExportDataProps) {
     }
   };
 
+  const downloadFlamegraphDotCom = () => {
+    if (!props.exportFlamegraphDotCom) {
+      return;
+    }
+
+    // TODO additional check this won't be needed once we use strictNullChecks
+    if (props.exportFlamegraphDotCom) {
+      const { flamebearer } = props;
+
+      const filename = `${getFilename(
+        flamebearer.metadata.appName,
+        flamebearer.metadata.startTime,
+        flamebearer.metadata.endTime
+      )}.png`;
+
+      // we don't upload to flamegraph.com directly due to potential CORS issues
+      fetch('/export', {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          name: filename,
+          profile: btoa(JSON.stringify(flamebearer)),
+          type: 'application/json',
+        }),
+      })
+        .then((response) => handleResponse(response))
+        .then((json) => {
+          const dlLink = document.createElement('a');
+          dlLink.target = '_blank';
+          dlLink.href = json.url;
+
+          document.body.appendChild(dlLink);
+          dlLink.click();
+          document.body.removeChild(dlLink);
+        })
+        .catch((e) => {
+          console.error(e);
+          // TODO: show error to user via notifications
```

Comment:
@eh-am curious how you think I should implement this. I was thinking about passing some callback (`errorHandler`) from the parent component to this component. That way we preserve its reusability.

Reply:
This is fine for now, but at some point the best of handling this is to lift this up to receive a `exportFlamegraphComFunction`, since  each environment will have a different way of handling this, eg grafana.